frontend/src/components/ExplorerCard/utils.ts

```diff
@@ -0,0 +1,177 @@
+import axios from 'axios';
+import { SOMETHING_WENT_WRONG } from 'constants/api';
+import { initialQueriesMap } from 'constants/queryBuilder';
+import {
+	queryParamNamesMap,
+	querySearchParams,
+} from 'constants/queryBuilderQueryNames';
+import { mapQueryDataFromApi } from 'lib/newQueryBuilder/queryBuilderMappers/mapQueryDataFromApi';
+import { isEqual } from 'lodash-es';
+
+import {
+	DeleteViewHandlerProps,
+	GetViewDetailsUsingViewKey,
+	IsQueryUpdatedInViewProps,
+	SaveViewHandlerProps,
+	UpdateQueryHandlerProps,
+} from './types';
+
+export const getViewDetailsUsingViewKey: GetViewDetailsUsingViewKey = (
+	viewKey,
+	data,
+) => {
+	const selectedView = data?.find((view) => view.uuid === viewKey);
+	if (selectedView) {
+		const { compositeQuery, name, uuid } = selectedView;
+		const query = mapQueryDataFromApi(compositeQuery);
+		return { query, name, uuid };
+	}
+	return undefined;
+};
+
+export const updateQueryHandler = async ({
+	updateViewAsync,
+	compositeQuery,
+	viewName,
+	viewKey,
+	sourcePage,
+	extraData,
+	setIsQueryUpdated,
+	notifications,
+}: UpdateQueryHandlerProps): Promise<void> => {
+	try {
+		await updateViewAsync({
+			compositeQuery,
+			viewKey,
+			extraData,
+			sourcePage,
+			viewName,
+		});
+		setIsQueryUpdated(false);
+		notifications.success({
+			message: 'View Updated Successfully',
+		});
+	} catch (err) {
+		notifications.error({
+			message: axios.isAxiosError(err) ? err.message : SOMETHING_WENT_WRONG,
+		});
+	}
+};
+
+export const isQueryUpdatedInView = ({
+	viewKey,
+	data,
+	stagedQuery,
+}: IsQueryUpdatedInViewProps): boolean => {
+	const currentViewDetails = getViewDetailsUsingViewKey(viewKey, data);
+	if (!currentViewDetails) {
+		return false;
+	}
+	const { query } = currentViewDetails;
+
+	// Omitting id from aggregateAttribute and groupBy
+	const updatedCurrentQuery = {
+		...stagedQuery,
+		builder: {
+			...stagedQuery?.builder,
+			queryData: stagedQuery?.builder.queryData.map((queryData) => {
+				const newAggregateAttribute = queryData.aggregateAttribute;
+				delete newAggregateAttribute.id;
```

Comment:
https://javascript.plainenglish.io/avoid-the-delete-keyword-in-javascript-87ff2a47f26c

Reply:
Okay,